ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -747,11 +748,33 @@ runThreadNetwork ThreadNetworkArgs
           -- traces the node's local events other than those from the -- ChainDB
           tracers = instrumentationTracers <> nullDebugTracers
 
+      let -- use a backoff delay of exactly one slot length for the following
+          -- reasons
+          --
+          -- o It gives the node a chance to sync some blocks so that it will
+          --   eventually not need to backoff
+          --
+          -- o It maintains the invariant that the node's activities all happen "
+          --   during " a slot onset
+          --
+          -- o It avoids causing the node to miss a slot it could have
+          --   nominally lead. EG If we used a backoff of two slot durations,
+          --   then it might have synced during the first slot and then been
+          --   able to productively lead the second slot had it not still been
+          --   asleep.
+          --
+          -- o We assume a node will only backoff when it joins late and only
+          --   until it syncs enough of the net's existing common prefix.
+          --
+          -- TODO The current slot duration will depend on the era of the join
```

Comment:
This TODO is the remaining work for Issue #2029 -- making the backoffdelay oracular in that it knows when the slot duration changes.

Reply:
The TODO is resolved in the latest commits on this PR (`OracularClock` et al).